dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -73,7 +73,12 @@ public abstract class KernelFunction
     /// <summary>
     /// Gets the prompt execution settings.
     /// </summary>
-    internal IReadOnlyDictionary<string, PromptExecutionSettings>? ExecutionSettings { get; }
+    /// <remarks>
+    /// The execution settings are used to configure the behavior of the function when calling a LLM.
+    /// The currently configured instance of <@see cref="IAIServiceSelector"/> from the <see cref="Kernel"/> is used
+    /// to determine the <see cref="IAIService"/> and <see cref="PromptExecutionSettings"/> to use for the function.
+    /// </remarks>
+    public IReadOnlyDictionary<string, PromptExecutionSettings>? ExecutionSettings { get; }
```

Comment:
This now makes KernelFunction partially mutable because PromptExecutionSettings is mutable, right?

Reply:
@stephentoub I also created a branch where I made a defensive copy of the execution settings so that `KernelFunction` remains immutable. Here's the draft PR https://github.com/microsoft/semantic-kernel/pull/4531.

Interested to hear your thoughts on this approach?

Reply:
Thanks. Left a comment on that PR.